server/src/main/java/org/apache/druid/server/metrics/MetricsModule.java

```diff
@@ -110,9 +108,10 @@ public MonitorScheduler getMonitorScheduler(
   {
     List<Monitor> monitors = new ArrayList<>();
     // HACK: when ServiceStatusMonitor is the first to be loaded, it introduces a circular dependency between
-    // CliPeon.runTask and CliPeon.getDataSourceFromTask/CliPeon.getTaskIDFromTask. The reason for this is unclear
-    // but by injecting DataSourceTaskIdHolder early this cycle is avoided.
-    injector.getInstance(DataSourceTaskIdHolder.class);
+    // CliPeon.runTask and TaskHolder.getDataSource()/TaskHolder.getTaskId(). The reason for this is unclear
+    // but by injecting TaskPropertiesHolder early this cycle is avoided.
```

Comment:
Is this true even after the changes in this PR?

Reply:
Ah yes, good point, they're no longer required. I've removed the changes originally included in https://github.com/apache/druid/pull/16140 and added some tests that would otherwise fail if we remove this "hack".